server/jetstream_cluster.go

```diff
@@ -7296,24 +7297,49 @@ func (s *Server) jsClusteredMsgDeleteRequest(ci *ClientInfo, acc *Account, mset
 	s.sendAPIResponse(ci, acc, subject, reply, string(rmsg), s.jsonResponse(resp))
 }
 
+// saPool pools stream assignment structs for encoding.
+var saPool = sync.Pool{
+	New: func() any {
+		return &streamAssignment{}
+	},
+}
+
+func saPoolPut(sa *streamAssignment) {
+	ciPool.Put(sa.Client)
+	sa.Client = nil
+	saPool.Put(sa)
+}
+
 func encodeAddStreamAssignment(sa *streamAssignment) []byte {
+	csa := saPool.Get().(*streamAssignment)
+	defer saPoolPut(csa)
+	*csa = *sa
+	csa.Client = csa.Client.forProposal()
 	var bb bytes.Buffer
 	bb.WriteByte(byte(assignStreamOp))
-	json.NewEncoder(&bb).Encode(sa)
+	json.NewEncoder(&bb).Encode(&csa)
 	return bb.Bytes()
 }
 
 func encodeUpdateStreamAssignment(sa *streamAssignment) []byte {
+	csa := saPool.Get().(*streamAssignment)
+	defer saPoolPut(csa)
+	*csa = *sa
```

Comment:
Maybe capture client info here in local variable from sa, and nil out before this line and put it back after?

Reply:
I don't think we need do, `csa` is a copy and therefore the `csa.Client = ...` line is operating on the copy. We don't modify the original SA.